src/passes/MergeBlocks.cpp

```diff
@@ -700,6 +684,12 @@ struct MergeBlocks
       outer = optimize(curr, curr->operands[i], outer);
     }
   }
+
+  void visitFunction(Function* curr) {
+    if (refinalize) {
+      ReFinalize().walkFunctionInModule(curr, getModule());
+    }
+  }
```

Comment:
Does this run repeatedly? If `visitFunction` runs at the end of the pass just once, how do we take advantage of the propagation?

And if unreachable child is fine to optimize, if the expression is unreachable, do we still bail out? (If so, where?)

Reply:
This only runs once, since `visitFunction` only happens once after each function.

We don't benefit from propagating unreachability inside the pass. The benefit in the PR is that we no longer give up when we see an unreachable child. (There might also be benefits to propagating it in the middle of the pass, separately - I hadn't considered that. But we do run the pass more than once, so I am not worried we are missing out on much.)

> And if unreachable child is fine to optimize, if the expression is unreachable, do we still bail out? (If so, where?)

After this PR we shouldn't be bailing out due to unreachability anywhere (unless I've missed some part of the code).